open-grant-proposals/Sector Repair Tool.md

```diff
@@ -0,0 +1,85 @@
+A: Under the circumstance of lacking raw data, recovering sectors with actual transaction data can not be acchived.
+
+## Deliverables
+
+We plan to create a sector repair tool, and the approaches will be one of the three options below: 
+1. Redo the sector-related functional code and release it with Lotus-Miner new version updates
+2. Redo the sector-independent program code and Lotus-redo project code
+3. Redo sector-independent program executables (binary), Lotus-redo (binary)
+
+## Development Roadmap
+
+No.  ll                                                                                      Milestone                                                                                      ll                time                ll      budget     ll
+1         Lotus-Miner sector state machine adds redo state, the corresponding redo state flow rules and flow logic.                             2 persons/ per week             $3,700
+2         Lotus-Miner sector and storage relationship management (local) reworks the storage information of the redone sector.        2 persons/ per week              $3,750
+3         Compatible processing of Lotus-worker to redo sectors.                                                                                                         1 person/ per week                $1,500
+4         Lotus-redo independent redo tool.                                                                                                                                         1 person/ per week                $1,500
+5         32G/64G sector redo function test.                                                                                                                                         1 person/week                       $1,500
+
+## Total Budget Requested
+
+Total budget is about $12,000.
+
+## Maintenance and Upgrade Plans
+
+We will keep an eye on Filecoin version updates, and we will provide corresponding maintenance and upgrades.
+
+# Team
+
+## Team Members
+
+- Team Member 1: Xianrui Zeng
+- Team Member 2: Caiyu Huang
+- Team Member 3: Gexue Zhang
+- Team Member 4: Bo Peng
+- Team Member 5: Teng Li
+
+## Team Member LinkedIn Profiles
+
+- Team Member 1 LinkedIn profile: https://www.linkedin.com/in/xianruizengb38a16218/
+- Team Member 2 LinkedIn profile: https://www.linkedin.com/in/caiyu-huang-455574218/
+- Team Member 3 LinkedIn profile: https://www.linkedin.com/in/gexue-zhang-485bb37b/
+- Team Member 4 LinkedIn profile: https://www.linkedin.com/in/bo-peng-021398219/
+- Team Member 5 LinkedIn profile: https://www.linkedin.com/in/teng-li-44939a219/
+
+## Team Website
+
+None.
+
+## Relevant Experience
+
+Please describe (in words) your team's relevant experience, and why you think you are the right team to build this project. You can cite your team's prior experience in similar domains, doing similar dev work, individual team members' backgrounds, etc.
```

Comment:
Please let us know if you  have any experience with filecoin, especially  running lotus miner. If you have contributed to lotus code base before - please linked prs

Reply:
We have our own lotus version which has been maintaining since the beginning. Its features include:
1.Refactoring the task distribution mechanism（Only on the same host, the task flows from AP to C1）
2.A supplement to flow logic of sector state (sectors can enter removed state under any other state, which is convenient to maintain. )
3.Sector data is written in the storage directly from the worker host server, instead of using HTTP network transmission between worker and miner. 
These features have never been uploaded to the lotus official staff, and our version has not been uploaded on Github. 

We have been following the IPFS project continuously from the beginning. As soon as the Filecoin mainnet went alive in October 2020, we started preparing to build nodes. On December 1st 2020, We ran our first node, F092887. Currently, the nodes we are building and operating are the ones below: 
f0121810
f0136808
f092887
f0110808
f0704940
f0704966
f0728817
f0811662
f0705136
f0500685
f01040516
f01043666
f01074227
We can modify the node labels as authentication if necessary.